cart.go

```diff
@@ -495,3 +473,31 @@ func mutateURL(original string, mutate bool) string {
 
 	return safe.String()
 }
+
+func httpClient(url, token string) []byte {
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req.Header.Set("Circle-Token", token)
+	req.Header.Set("Accept", "application/json")
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if res.StatusCode != http.StatusOK {
+		log.Fatalf("http: remote server responded %s (check http://status.circleci.com)", res.Status)
+	}
+
+	defer res.Body.Close()
+	body, readErr := ioutil.ReadAll(res.Body)
```

Comment:
1. This should use `io.ReadAll` (`ioutil` package deprecated).
2. Probably want to wrap `res.Body` in an `io.LimitedReader` to avoid resource exhaustion.

Reply:
Fixed 1 but on point 2 I don't think I understand. I'm not sure what to use as a sensible limit because some users will end up having too large of a file.

Reply:
Right. Honestly, it should just stream the read to the file and not buffer the whole artifact in memory. The `httpGet` function should probably just return the `http.Response` or the response body, and then the caller can use an `io.Copy` to stream the body to the file on disk.

Reply:
Okay, it should be cleaned up now @ydnar when you have time.